src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -1267,4 +1270,43 @@ final case class AgreementApiServiceImpl(
       getAgreementEServicesResponse[CompactEServices](operationLabel)(getAgreementEServices200)
     }
   }
+
+  override def archiveAgreement(agreementId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route = authorize(ADMIN_ROLE) {
+    val operationLabel = s"Archiving agreement $agreementId"
+    logger.info(operationLabel)
+
+    def archive(agreement: AgreementManagement.Agreement): Future[AgreementManagement.Agreement] = for {
+      uid       <- getUidFutureUUID(contexts)
+      agreement <- agreementManagementService.updateAgreement(
+        agreement.id,
+        AgreementManagement.UpdateAgreementSeed(
+          state = AgreementManagement.AgreementState.ARCHIVED,
+          certifiedAttributes = agreement.certifiedAttributes,
+          declaredAttributes = agreement.declaredAttributes,
+          verifiedAttributes = agreement.verifiedAttributes,
+          suspendedByConsumer = agreement.suspendedByConsumer,
+          suspendedByProducer = agreement.suspendedByProducer,
+          suspendedByPlatform = agreement.suspendedByPlatform,
+          stamps = agreement.stamps.copy(archiving = Stamp(uid, offsetDateTimeSupplier.get()).some)
+        )
+      )
+    } yield agreement
+
+    val result: Future[Agreement] = for {
+      requesterOrgId <- getOrganizationIdFutureUUID(contexts)
+      agreementUUID  <- agreementId.toFutureUUID
+      agreement      <- agreementManagementService.getAgreementById(agreementUUID)
+      _              <- assertRequesterIsProducer(requesterOrgId, agreement)
```

Comment:
As per story description
```suggestion
      _              <- assertRequesterIsConsumer(requesterOrgId, agreement)
```